include/deal.II/lac/linear_operator.h

```diff
@@ -791,6 +791,39 @@ identity_operator(const std::function<void(Range &, bool)> &reinit_vector)
 /**
  * @relates LinearOperator
  *
+ * Return a LinearOperator that is the identity of the vector space @p Range.
+ *
+ * The function takes a LinearOperator @p op and uses its range initializer
+ * to create an identiy operator. In contrast to the function above, this
+ * function also ensures that the underlying Payload matches that of the
+ * input @p op.
+ *
+ * @ingroup LAOperators
+ */
+template <typename Range, typename Domain, typename Payload>
+LinearOperator<Range, Domain, Payload>
+identity_operator(const LinearOperator<Range, Domain, Payload> &op)
+{
+  const LinearOperator<Range, Domain, Payload> id_op
+    = identity_operator<Range,Payload>(op.reinit_range_vector);
+  LinearOperator<Range, Domain, Payload> return_op (
+    op.identity_payload()
+  );
+
+  return_op.reinit_range_vector = id_op.reinit_range_vector;
+  return_op.reinit_domain_vector = id_op.reinit_domain_vector;
+  return_op.vmult = id_op.vmult;
+  return_op.vmult_add = id_op.vmult_add;
```

Comment:
Sorry, I'm struggling to understand what you mean here. Have you perhaps misread `id_op.vmult_add` as `op.vmult_add`? 

Up to this point the payload has populated the specialised `vmult` etc. function definitions, but those for the `Range` vector as still uninitialised. So instead of defining them explicitly here (and duplicating whats the other `identity_operator` function does), I simply create an instance of an `identity_operator` with no payload defined and copy over its `vmult` etc. definitions.